crates/manifest/src/schema/v2.rs

```diff
@@ -404,4 +587,147 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn test_check_disjoint() {
+        for (a, b) in [
+            ("foo:bar@0.1.0", "foo:bar@0.2.0"),
+            ("foo:bar/baz@0.1.0", "foo:bar/baz@0.2.0"),
+            ("foo:bar/baz@0.1.0", "foo:bar/bub@0.1.0"),
+            ("foo:bar@0.1.0", "foo:bar/bub@0.2.0"),
+            ("foo:bar@1.0.0", "foo:bar@2.0.0"),
+            ("foo:bar@0.1.0", "foo:bar@1.0.0"),
+            ("foo:bar/baz", "foo:bar/bub"),
+            ("foo:bar/baz@0.1.0-alpha", "foo:bar/baz@0.1.0-beta"),
+        ] {
+            let a: DependencyPackageName = a.parse().unwrap();
+            let b: DependencyPackageName = b.parse().unwrap();
+            ComponentDependencies::check_disjoint(&a, &b).unwrap(); // TODO: should method on depednecny name
+        }
+
+        for (a, b) in [
+            ("foo:bar@0.1.0", "foo:bar@0.1.1"),
+            ("foo:bar/baz@0.1.0", "foo:bar@0.1.0"),
+            ("foo:bar/baz@0.1.0", "foo:bar@0.1.0"),
+            ("foo:bar", "foo:bar@0.1.0"),
+            ("foo:bar@0.1.0-pre", "foo:bar@0.1.0-pre"),
+        ] {
+            let a: DependencyPackageName = a.parse().unwrap();
+            let b: DependencyPackageName = b.parse().unwrap();
+            assert!(
+                ComponentDependencies::check_disjoint(&a, &b).is_err(),
+                "{a} should conflict with {b}",
+            );
+        }
+    }
+
+    #[test]
+    fn test_validate_dependencies() {
+        // Specifying a dependency name as a plain-name without a package is an error
+        assert!(ComponentDependencies::deserialize(toml! {
+            "foo-bar" = "0.1.0"
```

Comment:
This might be a case where `foo` and `bar` are unhelpful identifiers.  I've seen so many `foo:bar` today that it took me a _lot_ of staring at this to realise this was hyphenated rather than coloned, and hence why it was bad.

E.g. `my-name`, `my-package:my-name`, `ns:name/itf`, etc.